core/src/main/java/org/apache/calcite/rel/metadata/BuiltInMetadata.java

```diff
@@ -105,6 +105,44 @@ interface Handler extends MetadataHandler<UniqueKeys> {
     }
   }
 
+  /**
+   * Metadata about which columns have foreign keys.
+   */
+  public interface ForeignKeys extends Metadata {
+    MetadataDef<ForeignKeys> DEF =
+        MetadataDef.of(ForeignKeys.class, ForeignKeys.Handler.class,
+            BuiltInMethod.FOREIGN_KEYS.method);
+
+    /**
+     * Determines the list of foreign keys for this expression. Foreign keys are
+     * represented as an {@link org.apache.calcite.util.ImmutableBitSet}, where
+     * each bit position represents the column ordinal is foreign key.
```

Comment:
I am not sure I understand this sentence, is it something like "where each bit position represents the column ordinal for the foreign key" what you mean or similar?

I mean, I get that the bit position is encoding the ordinal position of each foreign key, but I am not sure about the wording.

Reply:
You are right, the getForeignKeys method is similar to getUniqueKeys.   The result of getForeignKeys method represent the foreign keys pos above the current relNode, and the index is "0-based".   My comment was misunderstood, I'll edit it.

Reply:
I am not a native speaker, Thanks for your comment grammar check.